mbta.rb

```diff
@@ -1,25 +1,82 @@
 # Your solution here
 class Subway
 
-  def initialize 
-    red = ['South Station', 'Park Street', 'Kendall', 'Central', 'Harvard', 'Porter', 'Davis', 'Alewife']
-    green =  ['Government Center', 'Park Street', 'Boylston', 'Arlingrton', 'Copley', 'Hynes', 'Kenmore']
-    orange = ['North Station', 'Haymarket', 'Park Street', 'Downtown Crossing', 'chinatown', 'Back Bay', 'Forest Hills']
-    # Declare stations array for each line
-    park_street = Station.new('Park Street')
+  def initialize
+    lines_stations = [
+      ['South Station', 'Park Street', 'Kendall', 'Central', 'Harvard', 'Porter', 'Davis', 'Alewife'],
+      ['Government Center', 'Park Street', 'Boylston', 'Arlingrton', 'Copley', 'Hynes', 'Kenmore'],
+      ['North Station', 'Haymarket', 'Park Street', 'Downtown Crossing', 'chinatown', 'Back Bay', 'Forest Hills']
+    ]
+          
+    @park_street = Station.new('Park Street')
 
-    self.create_line_stations(red, 'Red', park_street)
-
-
-    red = Line.new()
+    @line_colors = ['Red', 'Green', 'Orange']
+    @lines = []
+    lines_stations.length.times do |i|
+      @lines.push(self.create_line_stations(lines_stations[i], @line_colors[i], @park_street))
+    end
 
   end
   def stops_between_stations(start_line, start_station, end_line, end_station)
     
+    start_line = self.getLineObject(@lines, start_line)
+    end_line = self.getLineObject(@lines, end_line)
```

Comment:
```ruby
   start_line = self.instance_variable_get("@#{start_line}")
   end_line = self.instance_variable_get("@#{end_line}")
```
this will replace the string with the value of the instance variable with the same name, without having to create a method for it